Sources/ProjectDescription/Settings.swift

```diff
@@ -1,22 +1,63 @@
 import Foundation
 
+// MARK: - SettingValue
+
+public enum SettingValue: ExpressibleByStringLiteral, ExpressibleByArrayLiteral, Equatable, Codable {
+    case string(String)
+    case array([String])
+
+    public init(stringLiteral value: String) {
+        self = .string(value)
+    }
+
+    public init(arrayLiteral elements: String...) {
+        self = .array(elements)
+    }
+
+    public init(from decoder: Decoder) throws {
+        guard let singleValueContainer = try? decoder.singleValueContainer() else {
+            preconditionFailure("Unsupported container type")
+        }
+        if let value: String = try? singleValueContainer.decode(String.self) {
+            self = .string(value)
+            return
+        }
+        if let value: [String] = try? singleValueContainer.decode([String].self) {
+            self = .array(value)
+            return
+        }
+
+        preconditionFailure("Unsupported encoded type")
```

Comment:
This doesn't seem like a precondition failure, more like a fatalError

Reply:
Good point 👍